.github/workflows/publish.yml

```diff
@@ -3,9 +3,12 @@ name: publish_docs
 on:
   push:
     branches:
-    - main
+      - main
+      - 'release-*'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
```

Comment:
We need to determine how we intend to update documentation for versioned releases. I would suggest that we only sync documentation for a release when vX.Y.0 is tagged, but not any other patch release.

This means that documentation improvements can be immediately backported into a release by a PR to the docs folder. 

If we leave this as-is, that means:

* You cannot manually backport changes to a release folder because it would get overwritten if we did a patch release 
* You have to wait for a patch release to be created for updated release docs to get re-synced (if a patch release ever gets created at all)
* Backporting changes to multiple previous releases is more difficult, as you have to do it across one PR per release branch

Reply:
Leaving some thoughts here but I may be missing some context that is specific to the Agent repo and workflow, so this is more from a Tempo point of view.  

The tags and branch filters can independently trigger, so I think the workflow will handle the cases mentioned.

> You cannot manually backport changes to a release folder because it would get overwritten if we did a patch release

It's expected to build the next patch off of the existing release branch, and any docs updates would have already been committed there.   Could you give an example of what might get overwritten?

> You have to wait for a patch release to be created for updated release docs to get re-synced (if a patch release ever gets created at all)

A commit to the release branch should resync immediately, no tag or release needed.

> Backporting changes to multiple previous releases is more difficult, as you have to do it across one PR per release branch

This is true and was not a design goal.  I'm not familiar with this use case, or sure if it's common enough to address.  Since the website documentation is forked into subfolders, I'm not sure what automation would look like to sync them all off of 1 PR.  What did you have in mind?

Reply:
> It's expected to build the next patch off of the existing release branch, and any docs updates would have already been committed there. Could you give an example of what might get overwritten?

Right, I'm saying if copy changes are done to grafana/website so they appear immediately, and not synced to grafana/agent. The next patch release tag will re-sync the folder overriding any manual changes done. 

1. Release v0.100.0 gets cut, `v0.100` docs synced to grafana/website 
2. Copy changes manually made to grafana/website/docs/agent/v0.100 
3. Release v0.100.1 gets cut, `v0.100` docs re-synced to grafana/website 
4. Previous copy changes get lost because they were not made to grafana/agent 

> A commit to the release branch should resync immediately, no tag or release needed.

That should only be true after we cut a stable release, which won't be true for every release branch (unfortunately). We create a release-X.Y for every X.Y.Z release, including release candidates.

> This is true and was not a design goal. I'm not familiar with this use case, or sure if it's common enough to address. Since the website documentation is forked into subfolders, I'm not sure what automation would look like to sync them all off of 1 PR. What did you have in mind?

My understanding is that documentation writers will manually backport changes to grafana/website. With the current flow, they're able to backport to multiple previous versions at once (since it's one giant directory tree). If the source of truth becomes repositories, then it becomes slightly more tedious. 

My understanding could be wrong, or it could also be that nobody would mind this change.

Reply:
FWIW, we document our current release process [here](https://github.com/grafana/agent/blob/main/docs/developer/releasing.md#release-branches).

Reply:
> copy changes are done to grafana/website

Ok yes, agree with everything you've said under that scenario. My understanding is that kind of activity is rare, and will be moreso after this automation.  We do want the upstream repositories to be the source of truth.

> backport to multiple previous versions at once
Agree, that scenario would not be supported.  I have not heard it mentioned so thinking it's a worthwhile tradeoff for the increased automation in general.  Definitely for Tempo and GET.

> single release-X.Y branch
We do the same.  I think I understand part of your concern in that under this setup the documentation would get to the website "early", before the final stable candidate is pushed.  However, the thoughts are that it's not a blocker for a few reasons:  (1) the time between starting a release (creating the branch) and the stable .0 tag is relatively short (a few days) (2) we can devise a tweak to the workflow later if needed (already have some rough ideas).

> won't be true for every release branch 
Got it, for Tempo that is not an issue.  I think we have a good base to build on here, but expect every project will have some customization.  Syncing on both tags and branch commits is already quite heavy-handed and more just to start with a generic solution and experimentation.

Reply:
> My understanding is that documentation writers will manually backport changes to grafana/website. With the current flow, they're able to backport to multiple previous versions at once (since it's one giant directory tree). If the source of truth becomes repositories, then it becomes slightly more tedious.

This isn't the preferred approach. The website should reflect the technical documentation stored in the product/project repo, with the website automatically reflecting that change. Thus, backporting should happen only in the product/project repo.

Reply:
Thanks for explaining!

Reply:
Backporting is relatively common in some products, not as frequent in others. We'd like to enable a consistent approach. And our aim is to not backport more than two minor releases back, similar to the backporting policy for security fixes.

Reply:
> Agree, that scenario would not be supported. I have not heard it mentioned so thinking it's a worthwhile tradeoff for the increased automation in general. Definitely for Tempo and GET.
> 
> >

I agree

Reply:
I think we've teased out some good issues here. @jdbaldry may also want to comment, as he's focused now on the docs build pipeline with the goal of creating more consistency and sharing practices.

Thank you for the dialog on this - I think it's definitely a step forward